### PR TITLE
New tests for CityScapes dataset

### PR DESCRIPTION
**Pull request by NicolasHug**
This PR ports the CityScapes tests to the new test infrastructure.

Addresses part of #3531

### CODE REVIEW

**Review by pmeier — COMMENTED** [review]
Looking good, but I think the new tests are covering less than the old ones. We don't seem to test for different `target_type`'s anymore. Could you also port these?

**Comment by NicolasHug** [comment]
Ah good catch, thanks. Should be OK now :)

File: test/test_datasets.py [hunk 1]
```diff
@@ -520,6 +475,104 @@ def inject_fake_data(self, tmpdir, config):
...
+        "color",
+        ["instance", "semantic"],
+    )
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(
+        mode=("fine",), split=("train", "test", "val"), target_type=TARGET_TYPES
+    ) + datasets_utils.combinations_grid(
+        mode=("coarse",),
+        split=("train", "train_extra", "val"),
+        target_type=TARGET_TYPES,
+    )
```
Comment (pmeier) [inline test/test_datasets.py:493]: Nit: this is hard to read

```suggestion
    ADDITIONAL_CONFIGS = (
        *datasets_utils.combinations_grid(
            mode=("fine",), split=("train", "test", "val"), target_type=TARGET_TYPES
        ),
        *datasets_utils.combinations_grid(
            mode=("coarse",),
            split=("train", "train_extra", "val"),
            target_type=TARGET_TYPES,
        ),
    )
```

File: test/test_datasets.py [hunk 2]
```diff
@@ -149,51 +149,6 @@ def test_fashionmnist(self, mock_download_extract, mock_check_integrity):
...
-                    self.assertTrue(isinstance(color_img, PIL.Image.Image))
-                    self.assertTrue(np.array(color_target).shape[2] == 4)
-
-                    polygon_dataset = torchvision.datasets.Cityscapes(
-                        root, split=split, target_type='polygon', mode=mode)
-                    polygon_img, polygon_target = polygon_dataset[0]
-                    self.assertTrue(isinstance(polygon_img, PIL.Image.Image))
-                    self.assertTrue(isinstance(polygon_target, dict))
-                    self.assertTrue(isinstance(polygon_target['imgHeight'], int))
-                    self.assertTrue(isinstance(polygon_target['objects'], list))
```
Comment (pmeier) [inline test/test_datasets.py:181]: A check like this is still missing. I suggest you add a new `test_feature_types_target_polygon()` and test the returned dictionary. To make this even more robust you might return the correct values in an info dict from `inject_fake_data` and also test them here.

File: test/test_datasets.py [hunk 3]
```diff
@@ -149,51 +149,6 @@ def test_fashionmnist(self, mock_download_extract, mock_check_integrity):
...
-                    for target_type in ['semantic', 'instance']:
-                        dataset = torchvision.datasets.Cityscapes(
-                            root, split=split, target_type=target_type, mode=mode)
-                        self.generic_segmentation_dataset_test(dataset, num_images=2)
-
-                    color_dataset = torchvision.datasets.Cityscapes(
-                        root, split=split, target_type='color', mode=mode)
-                    color_img, color_target = color_dataset[0]
-                    self.assertTrue(isinstance(color_img, PIL.Image.Image))
-                    self.assertTrue(np.array(color_target).shape[2] == 4)
```
Comment (pmeier) [inline test/test_datasets.py:173]: Same as above.

File: test/test_datasets.py [hunk 4]
```diff
@@ -149,51 +149,6 @@ def test_fashionmnist(self, mock_download_extract, mock_check_integrity):
...
-                    polygon_img, polygon_target = polygon_dataset[0]
-                    self.assertTrue(isinstance(polygon_img, PIL.Image.Image))
-                    self.assertTrue(isinstance(polygon_target, dict))
-                    self.assertTrue(isinstance(polygon_target['imgHeight'], int))
-                    self.assertTrue(isinstance(polygon_target['objects'], list))
-
-                    # Test multiple target types
-                    targets_combo = ['semantic', 'polygon', 'color']
-                    multiple_types_dataset = torchvision.datasets.Cityscapes(
-                        root, split=split, target_type=targets_combo, mode=mode)
```
Comment (pmeier) [inline test/test_datasets.py:186]: While we now check multiple `target_type`'s the old test also checks if the returned content is valid. I did something similar for `Caltech101`:

https://github.com/pytorch/vision/blob/19ad0bbc5e26504a501b9be3f0345381d6ba1efc/test/test_datasets.py#L415

Maybe we can do the same here. Thoughts?

File: test/datasets_utils.py [hunk 5]
```diff
@@ -734,7 +734,7 @@ def size(idx: int) -> Tuple[int, int, int]:
             return (num_channels, height, width)
 
     root = pathlib.Path(root) / name
-    os.makedirs(root)
+    os.makedirs(root, exist_ok=True)
```
Comment (pmeier) [inline test/datasets_utils.py:737]: I think this is fine. Could you add the same to `create_video_folder()` so they don't diverge?

**Review by pmeier — APPROVED** [review]
LGTM, thanks!

**Comment by NicolasHug** [comment]
Thanks for the reviews!

**Pull request merged**